contracts/modules/STO/USDTiered/USDTieredSTO.sol

```diff
@@ -195,30 +199,58 @@ contract USDTieredSTO is USDTieredSTOStorage, STO {
      */
-    function modifyAddresses(address payable _wallet, address _treasuryWallet, IERC20[] calldata _usdTokens) external {
+    function modifyAddresses(address payable _wallet, address _treasuryWallet, IERC20[] calldata _stableTokens) external {
         _onlySecurityTokenOwner();
-        _modifyAddresses(_wallet, _treasuryWallet, _usdTokens);
+        _modifyAddresses(_wallet, _treasuryWallet, _stableTokens);
     }
 
     /**
-     * @dev Modifies Oracle address.
-     *      By default, Polymath oracles are used but issuer can overide them using this function
-     *      Set _oracleAddress to 0x0 to fallback to using Polymath oracles
-     * @param _fundRaiseType Actual currency
-     * @param _oracleAddress address of the oracle
+     * @notice Modifies the oracle addresses and the denominated currency
+     * @dev If custom oracle addresses and denominated currency symbol are 0x0 then it will automatically
+     *      fallback to the Polymath oracle addresses and USD denominated currency symbol
+     * @param _customOracleAddresses Addresses of the oracles
+     * @param _denominatedCurrencySymbol Symbol of the Fiat currency used for denomination
      */
-    function modifyOracle(FundRaiseType _fundRaiseType, address _oracleAddress) external {
+    function modifyOracles(address[] calldata _customOracleAddresses, bytes32 _denominatedCurrencySymbol) external {
         _onlySecurityTokenOwner();
-        if (_fundRaiseType == FundRaiseType.ETH) {
-            customOracles[bytes32("ETH")][bytes32("USD")] = _oracleAddress;
-        } else {
-            require(_fundRaiseType == FundRaiseType.POLY, "Invalid currency");
-            customOracles[bytes32("POLY")][bytes32("USD")] = _oracleAddress;
+        // If Issuer wants to change the denominatedCurrency then they have to change the rates as well according the denominated currency
+        // It is advised to call modifyTiers just after calling modifyOracles from the dApp side.
+        if (_denominatedCurrencySymbol != denominatedCurrency)
+            _isSTOStarted();
+        _modifyOracles(_customOracleAddresses, _denominatedCurrencySymbol);
+    }
+
+    function _modifyOracles(address[] memory _customOracleAddresses, bytes32 _denominatedCurrencySymbol) internal {
+        if (_customOracleAddresses.length == 0 && _denominatedCurrencySymbol == bytes32(0)) {
+            denominatedCurrency = bytes32("USD");
+            delete customOracles[bytes32("ETH")][denominatedCurrency];
+            delete customOracles[bytes32("POLY")][denominatedCurrency];
+            oracleKeys[bytes32("ETH")][denominatedCurrency] = ETH_ORACLE;
+            oracleKeys[bytes32("POLY")][denominatedCurrency] = POLY_ORACLE;
+            emit SetOracles(denominatedCurrency, false);
+        }
+        else {
+            require(_denominatedCurrencySymbol != bytes32(0), "Invalid currency");
+            require(_customOracleAddresses.length == 2, "Invalid no. of oracles");
```

Comment:
I think we should allow issuers to supply only ETH oracle if they are raising in ETH only.

Reply:
If they are only raising in stable coin, we shouldn't require them to enter two oracle addresses